File: 1957-delete-characters-to-make-fancy-string/1957-delete-characters-to-make-fancy-string.py
class Solution:
    def makeFancyString(self, s: str) -> str:
        result = list()
        cnt = 0

        for index, letter in enumerate(s):
            if index == 0 or letter != s[index - 1]:
                result.append(letter)
                cnt = 1
            else:
                if cnt == 2:
                    continue
                else:
                    cnt += 1
                    result.append(letter)

        return ''.join(result)

File: 1957-delete-characters-to-make-fancy-string/1957-delete-characters-to-make-fancy-string.cpp
class Solution {
public:
    string makeFancyString(string s) {
        string result = "";
        int cnt = 0;

        for (int i = 0; i < s.length(); i++) {
            if (i == 0 || s[i] != s[i - 1]) {
                result += s[i];
                cnt = 1;
            } else if (cnt == 2) {
                continue;
            } else {
                result += s[i];
                cnt++;
            }
        }

        return result;
    }
};